Extension/src/LanguageServer/Providers/callHierarchyProvider.ts

```diff
@@ -0,0 +1,199 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as vscode from 'vscode';
+import { DefaultClient } from '../client';
+import { processDelayedDidOpen } from '../extension';
+import { Position, Range, RequestType, TextDocumentIdentifier } from 'vscode-languageclient';
+import { makeVscodeRange } from '../utils';
+
+interface CallHierarchyItem {
+    /**
+     * The name of this item or symbol.
+     */
+    name: string;
+
+    /**
+     * The kind of this item.
+     */
+    kind: vscode.SymbolKind;
+
+    /**
+     * More detail for this item, e.g. the scope or class of a function.
+     */
+    detail: string;
+
+    /**
+     * The resource identifier of this item.
+     */
+    uri: string;
+
+    /**
+     * The range enclosing this symbol not including leading/trailing whitespace but everything else, e.g. comments and code.
+     */
+    range: Range;
+
+    /**
+     * The range that should be selected and revealed when this symbol is being picked, e.g. the name of a function.
+     * Must be contained by the `CallHierarchyItem.range`.
+     */
+    selectionRange: Range;
+}
+
+interface CallHierarchyParams {
+    textDocument: TextDocumentIdentifier;
+    position: Position;
+}
+
+interface CallHierarchyItemResult {
+    item?: CallHierarchyItem;
+
+    /**
+     * If a request is cancelled, `succeeded` will be undefined.
+     */
+    succeeded: boolean;
+}
+
+interface CallHierarchyCallsItem {
+    /**
+     * For CallHierarchyIncomingCall or calls to, this is the item that makes the call.
+     * For CallHierarchyOutgoingCall or calls from, this is the item that is called.
+     */
+    item: CallHierarchyItem;
+
+    /**
+     * For CallHierarchyIncomingCall or calls to, this is the range at which the call appears.
+     * For CallHierarchyOutgoingCall or calls from, this is the range at which this item is called.
+     */
+    fromRanges: Range[];
+}
+
+interface CallHierarchyCallsItemResult {
+    calls: CallHierarchyCallsItem[];
+}
+
+const CallHierarchyItemRequest: RequestType<CallHierarchyParams, CallHierarchyItemResult, void> =
+    new RequestType<CallHierarchyParams, CallHierarchyItemResult, void>('cpptools/prepareCallHierarchy');
+
+const CallHierarchyCallsToRequest: RequestType<CallHierarchyParams, CallHierarchyCallsItemResult, void> =
+new RequestType<CallHierarchyParams, CallHierarchyCallsItemResult, void>('cpptools/callHierarchyCallsTo');

```

Comment:
Can you run Format Document? There are 4 cases where lines are missing indentation.

Reply:
Actually, it looks like lint complains about the other 3 cases, so this is the only case.

Reply:
From looking at https://github.com/typescript-eslint/typescript-eslint/issues/1824 it sounds like we may want to disable the linter indent rule.

Reply:
The other cases results in a yarn lint error. So Only the first case can be formatted. I'll update the yarn lint.